program-bpf-rust/src/lib.rs

```diff
@@ -0,0 +1,242 @@
+#![no_std]
+
+extern crate serde;
+#[macro_use]
+extern crate serde_derive;
+#[cfg(not(test))]
+extern crate solana_sdk_bpf_no_std;
+extern crate solana_sdk_bpf_utils;
+
+// required to use vec! in the program_command.h tests
+#[cfg(test)]
+#[macro_use]
+extern crate alloc;
+
+mod dashboard;
+mod game;
+mod program_command;
+mod program_state;
+mod result;
+mod simple_serde;
+
+use program_command::Command;
+use program_state::State;
+use result::{ProgramError, Result as ProgramResult};
+use simple_serde::SimpleSerde;
+use solana_sdk_bpf_utils::entrypoint::*;
+use solana_sdk_bpf_utils::{entrypoint, info};
+
+fn get_current_slot(account: &SolKeyedAccount) -> u64 {
+    #[allow(clippy::cast_ptr_alignment)]
+    unsafe {
+        *(&account.data[0] as *const u8 as *const u64)
```

Comment:
🤕 Can we hide this `unsafe` in the sdk?  (maybe as a follow-up)

Reply:
I'd like to provide standard utilities to interact with the syscalls so users don't have to do this garbage themselves

Reply:
Cool.  So you'll swing back here later when that exists to clean this up?

Reply:
https://github.com/solana-labs/solana/issues/4981

Reply:
get_current_slot() must return a Result<> or an Option<> for the case where the keyed account is of the wrong type, right?

Reply:
Right, its hacked in right now, the SysCall utilities will handle this correctly.